src/neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -0,0 +1,200 @@
+        {
+            return snapshot[CreateStorageKey(Prefix_CurrentBlock)].GetInteroperable<HashIndexState>().Index;
+        }
+
+        public bool ContainsBlock(DataCache snapshot, UInt256 hash)
+        {
+            return snapshot.Contains(CreateStorageKey(Prefix_Block).Add(hash));
+        }
+
+        public bool ContainsTransaction(DataCache snapshot, UInt256 hash)
+        {
+            return snapshot.Contains(CreateStorageKey(Prefix_Transaction).Add(hash));
+        }
+
+        public TrimmedBlock GetTrimmedBlock(DataCache snapshot, UInt256 hash)
+        {
+            StorageItem item = snapshot.TryGet(CreateStorageKey(Prefix_Block).Add(hash));
+            if (item is null) return null;
+            return item.Value.AsSerializable<TrimmedBlock>();
+        }
+
+        [ContractMethod(0_01000000, CallFlags.ReadStates)]
+        private TrimmedBlock GetBlock(DataCache snapshot, byte[] indexOrHash)
+        {
+            UInt256 hash;
+            if (indexOrHash.Length < UInt256.Length)
+                hash = GetBlockHash(snapshot, (uint)new BigInteger(indexOrHash));
+            else if (indexOrHash.Length == UInt256.Length)
+                hash = new UInt256(indexOrHash);
+            else
+                throw new ArgumentException(null, nameof(indexOrHash));
+            if (hash is null) return null;
+            TrimmedBlock block = GetTrimmedBlock(snapshot, hash);
+            if (block is null || !IsTraceableBlock(snapshot, block.Index)) return null;
+            return block;
+        }
+
+        public Block GetBlock(DataCache snapshot, UInt256 hash)
+        {
+            TrimmedBlock state = GetTrimmedBlock(snapshot, hash);
+            if (state is null) return null;
+            return new Block
+            {
+                Version = state.Version,
+                PrevHash = state.PrevHash,
+                MerkleRoot = state.MerkleRoot,
+                Timestamp = state.Timestamp,
+                Index = state.Index,
+                NextConsensus = state.NextConsensus,
+                Witness = state.Witness,
+                ConsensusData = state.ConsensusData,
+                Transactions = state.Hashes.Skip(1).Select(p => GetTransaction(snapshot, p)).ToArray()
+            };
+        }
+
+        public Block GetBlock(DataCache snapshot, uint index)
+        {
+            UInt256 hash = GetBlockHash(snapshot, index);
+            if (hash is null) return null;
+            return GetBlock(snapshot, hash);
+        }
+
+        public Header GetHeader(DataCache snapshot, UInt256 hash)
+        {
+            return GetTrimmedBlock(snapshot, hash)?.Header;
+        }
+
+        public Header GetHeader(DataCache snapshot, uint index)
+        {
+            UInt256 hash = GetBlockHash(snapshot, index);
+            if (hash is null) return null;
+            return GetHeader(snapshot, hash);
+        }
+
+        public Transaction GetTransaction(DataCache snapshot, UInt256 hash)
```

Comment:
This should be changed to return TransactionState. RPC needs TransactionState data.

Reply:
Done